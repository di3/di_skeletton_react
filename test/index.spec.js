import React from 'react';
import index from 'index';
import requestFake, { set, reset } from 'di_request_fake';

beforeEach(() => {
  jest.doMock('di_request', (settings) => requestFake(settings));
  reset();
  set([
    {data: {title: "test"}, status: 200}
  ]);
});

test('initial snapshots with timeout', async (done) => {
  var div = document.createElement('div');
  div.id = "SKELETON";
  document.body.appendChild(div);
  index.create({container: "SKELETON", name: "skeleton"});
  expect.assertions(2);
  expect(div).toMatchSnapshot();
  await setTimeout(() => {
    expect(div).toMatchSnapshot()
    done();
  }, 1100);
});
test('initial snapshots with async timeout', async (done) => {
  var div = document.createElement('div');
  div.id = "SKELETON2";
  document.body.appendChild(div);
  index.create({container: "SKELETON2", name: "skeleton"});
  expect.assertions(2);
  expect(div).toMatchSnapshot();
  await setTimeout(() => {
    expect(div).toMatchSnapshot();
    done();
  }, 3100);
});
test('responsive snapshots with timeout', async (done) => {
  var div = document.createElement('div');
  div.id = "SKELETON3";
  document.body.appendChild(div);
  index.create({container: "SKELETON3", name: "skeleton"});

  // Change the viewport to 500px.
  global.innerWidth = 500;

  // Trigger the window resize event.
  global.dispatchEvent(new Event('resize'));
  expect.assertions(2);
  expect(div).toMatchSnapshot();

  await setTimeout(() => {
    expect(div).toMatchSnapshot();
    done();
  }, 1100);
});
