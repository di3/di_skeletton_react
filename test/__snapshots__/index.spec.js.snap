// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial snapshots with async timeout 1`] = `
<div
  id="SKELETON2"
>
  <div
    style="font-family: Monospace; width: 200px; display: block; margin: auto; font-size: 250%; text-align: center; color: rgb(86, 47, 79);"
  >
    <span
      style="font-weight: bold; display: block;"
    >
      skeleton
    </span>
    <span
      style="font-weight: bold; display: block;"
    >
      skeleton
    </span>
    <span
      style="padding: 10px; display: block; font-size: 150%;"
    >
      0
    </span>
    <div
      style="font-family: Monospace; display: block; margin: auto; font-size: 120%; border: 4px solid #562f4f; background-color: rgb(150, 111, 143); color: rgb(86, 47, 79); font-weight: bold; border-radius: 3px;"
    >
      reset
    </div>
  </div>
</div>
`;

exports[`initial snapshots with async timeout 2`] = `
<div
  id="SKELETON2"
>
  <div
    style="font-family: Monospace; width: 200px; display: block; margin: auto; font-size: 250%; text-align: center; color: rgb(86, 47, 79);"
  >
    <span
      style="font-weight: bold; display: block;"
    >
      skeleton
    </span>
    <span
      style="font-weight: bold; display: block;"
    >
      skeleton
    </span>
    <span
      style="padding: 10px; display: block; font-size: 150%;"
    >
      12
    </span>
    <div
      style="font-family: Monospace; display: block; margin: auto; font-size: 120%; border: 4px solid #562f4f; background-color: rgb(150, 111, 143); color: rgb(86, 47, 79); font-weight: bold; border-radius: 3px;"
    >
      reset
    </div>
  </div>
</div>
`;

exports[`initial snapshots with timeout 1`] = `
<div
  id="SKELETON"
>
  <div
    style="font-family: Monospace; width: 200px; display: block; margin: auto; font-size: 250%; text-align: center; color: rgb(86, 47, 79);"
  >
    <span
      style="font-weight: bold; display: block;"
    >
      skeleton
    </span>
    <span
      style="font-weight: bold; display: block;"
    >
      skeleton
    </span>
    <span
      style="padding: 10px; display: block; font-size: 150%;"
    >
      0
    </span>
    <div
      style="font-family: Monospace; display: block; margin: auto; font-size: 120%; border: 4px solid #562f4f; background-color: rgb(150, 111, 143); color: rgb(86, 47, 79); font-weight: bold; border-radius: 3px;"
    >
      reset
    </div>
  </div>
</div>
`;

exports[`initial snapshots with timeout 2`] = `
<div
  id="SKELETON"
>
  <div
    style="font-family: Monospace; width: 200px; display: block; margin: auto; font-size: 250%; text-align: center; color: rgb(86, 47, 79);"
  >
    <span
      style="font-weight: bold; display: block;"
    >
      skeleton
    </span>
    <span
      style="font-weight: bold; display: block;"
    >
      skeleton
    </span>
    <span
      style="padding: 10px; display: block; font-size: 150%;"
    >
      2
    </span>
    <div
      style="font-family: Monospace; display: block; margin: auto; font-size: 120%; border: 4px solid #562f4f; background-color: rgb(150, 111, 143); color: rgb(86, 47, 79); font-weight: bold; border-radius: 3px;"
    >
      reset
    </div>
  </div>
</div>
`;

exports[`responsive snapshots with timeout 1`] = `
<div
  id="SKELETON3"
>
  <div
    style="font-family: Monospace; width: 200px; display: block; margin: auto; font-size: 250%; text-align: center; color: rgb(86, 47, 79);"
  >
    <span
      style="font-weight: bold; display: block;"
    >
      skeleton
    </span>
    <span
      style="font-weight: bold; display: block;"
    >
      skeleton
    </span>
    <span
      style="padding: 10px; display: block; font-size: 150%;"
    >
      0
    </span>
    <div
      style="font-family: Monospace; display: block; margin: auto; font-size: 120%; border: 4px solid #562f4f; background-color: rgb(150, 111, 143); color: rgb(86, 47, 79); font-weight: bold; border-radius: 3px;"
    >
      reset
    </div>
  </div>
</div>
`;

exports[`responsive snapshots with timeout 2`] = `
<div
  id="SKELETON3"
>
  <div
    style="font-family: Monospace; width: 200px; display: block; margin: auto; font-size: 180%; text-align: center; color: rgb(86, 47, 79);"
  >
    <span
      style="font-weight: bold; display: block;"
    >
      skeleton
    </span>
    <span
      style="font-weight: bold; display: block;"
    >
      skeleton
    </span>
    <span
      style="padding: 10px; display: block; font-size: 150%;"
    >
      6
    </span>
    <div
      style="font-family: Monospace; display: block; margin: auto; font-size: 120%; border: 4px solid #562f4f; background-color: rgb(150, 111, 143); color: rgb(86, 47, 79); font-weight: bold; border-radius: 3px;"
    >
      reset
    </div>
  </div>
</div>
`;
